### ISSUE TITLE
[Feature]: XORless transfer

### ISSUE THREAD

**Issue by Tieumsan** [sora-xor/sora2-network#278]
### As a
User

### I want to
Transfer tokens without having to think about swapping to XOR to pay for the transaction fee

### So that
I can participate in the SORA economy without having to think about getting the necessary XOR to pay for the transaction fee

### Preconditions
The user does not have XOR balance.

### Flow
1. The user sends tokenA to addressX
2. The system checks whether the address has XOR to pay for the TX fee
3. If the address does not have XOR to pay for the TX fee, the system automatically swaps tokens to get enough XOR to pay for the transaction fee
4. The system sends the transaction

### Questions
- How do we inform users that part of his tokens will be swapped to XOR to pay for the transfer ?
- What if the user wants to transfer all the freeBalance and does not have sufficient remaining funds to swap to pay for the TX fee ?
- How do we pay the TX fee for the swap ?
- Which token do we choose to swap to XOR for the TX fee ?
- Do we pay a transaction fee for swapping to XOR to get the required amount of XOR to pay for the transaction fee ? Or is it possible to make a transaction fee-less swap ?

**Comment by stefashkaa** [comment]
@Tieumsan #244

**Comment by Tieumsan** [comment]
@bgodlin please review the requirements and questions to be addressed this week with @vovac12 (priority = order of releases).

**Comment by Tieumsan** [comment]
@bgodlin when can we start working on this ?
cc. @vovac12

**Comment by bgodlin** [comment]
> @bgodlin when can we start working on this ? cc. @vovac12

After the release of the bridge

**Comment by Tieumsan** [comment]
> > @bgodlin when can we start working on this ? cc. @vovac12
> 
> After the release of the bridge

no, we have a deadline for this release and expecting it for 3 weeks already. Complete it by Monday

**Comment by Tieumsan** [comment]
@bgodlin Looks ok to me. Having a sequence diagram would help ensure the flow is understood though, along with error cases.
@vovac12 please review

**Comment by ra9mls** [comment]
For improving usability could we check this task? https://github.com/sora-xor/sora2-network/issues/760

**Comment by vovac12** [comment]
Added event `XorlessTransfer(asset_id: AssetId, sender: AccountId, receiver: AccountId, amount: Balance, additional_data: Bytes)`
